src/main/resources/avro/genotypephenotypemethods.avdl

```diff
@@ -0,0 +1,319 @@
+of a particular class of object (e.g. a specific gene or SNP).
+*/
+record ExternalIdentifierQuery {
+  array<org.ga4gh.models.ExternalIdentifier> ids;
+}
+
+/**
+Evidence for the phenotype association.
+*/
+record EvidenceQuery {
+  /** ECO or OBI is recommended */
+  union { null, org.ga4gh.models.OntologyTerm }  evidenceType = null;
+  union { null, string } description = null;  /*regex*/
+  union { null, array<org.ga4gh.models.ExternalIdentifier> }  externalIdentifiers = null; /* new field */
+}
+
+
+record GenotypeQuery {
+  /**
+  only those fields from feature that are `queryable`
+  */
+  union { null, string } name; /* new field, regex */
+  union { null, string } description; /* new field,regex */
+  union { null, string } featureSetId;
+  union { null, string } referenceName;
+  union { null, long } start;
+  union { null, long } end;
+  union { null, org.ga4gh.models.Strand } strand;
+  union { null, org.ga4gh.models.OntologyTerm } type; /* new field  */
+  union { null, org.ga4gh.models.OntologyTerm } featureType;
+  union { null, array<org.ga4gh.models.ExternalIdentifier> } externalIdentifiers = null; /* new field */
+}
+
+record SearchPhenotypeRequest {
+
+  /**
+  The `PhenotypeAssociationSet` to search.
+  */
+  string phenotypeAssociationSetId;
+
+  array<PhenotypeQuery> phenotypes = [];
+
+  /**
+  Specifies the maximum number of results to return in a single page.
+  If unspecified, a system default will be used.
+  */
+  union { null, int } pageSize = null;
+
+  /**
+  The continuation token, which is used to page through large result sets.
+  To get the next page of results, set this parameter to the value of
+  `nextPageToken` from the previous response.
+  */
+  union { null, string } pageToken = null;
+}
+
+/** This is the response from `POST /phenotype/search` expressed as JSON. */
+record SearchPhenotypeResponse {
+  /**
+  The list of matching FeaturePhenotypeInstances.
+  */
+  array<org.ga4gh.models.PhenotypeInstance> phenotypes = [];
+
+  /**
+  The continuation token, which is used to page through large result sets.
+  Provide this value in a subsequent request to return the next page of
+  results. This field will be empty if there aren't any additional results.
+  */
+  union { null, string } nextPageToken = null;
+}
+
+/**
+Gets a list of evidence accessible through the API.
+
+`POST /phenotype/search` must accept a JSON version of
+`SearchGenotypePhenotypeRequest` as the post body and will return a JSON version
+of `SearchGenotypePhenotypeResponse`.
+*/
+SearchPhenotypeResponse searchPhenotype(
+  /**
+  This request maps to the body of `POST /phenotype/search` as JSON.
+  */
+  SearchPhenotypeRequest request) throws GAException;
+
+
+
+
+/******************  /genotype/search  *********************/
+
+record SearchGenotypesRequest {
+
+  /**
+  The `PhenotypeAssociationSet` to search.
+  */
+  string phenotypeAssociationSetId;
+
+  array<GenotypeQuery> genotypes = [];
```

Comment:
Do we really need the list of query objects here? Why not put the fields that are in GenotypeQuery directly in here, and make a series of requests?